apps/web-giddh/src/app/settings/integration/institutions-list/institutions-list.component.ts

```diff
@@ -0,0 +1,156 @@
+    styleUrls: ['./institutions-list.component.scss'],
+    templateUrl: './institutions-list.component.html',
+    providers: [SettingIntegrationComponentStore],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+
+export class InstitutionsListComponent implements OnInit, OnDestroy {
+    /* This will hold local JSON data */
+    public localeData: any = {};
+    /* This will hold common JSON data */
+    public commonLocaleData: any = {};
+    /** Subject to release subscription memory */
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /** Hold institutions list */
+    public institutions: any[] = [];
+    /** Holds Store Institutions list API success state as observable*/
+    public institutionsList$: Observable<any> = this.componentStore.select(state => state.institutionList);
+    /** This will hold institutions request api form request */
+    public institutionsRequest: InstitutionsRequest = new InstitutionsRequest();
+    /** This will use for open window */
+    public openedWindow: Window | null = null;
+    /** Holds Store Save payment provider company API success state as observable*/
+    public createEndUserAgreementSuccess$: Observable<any> = this.componentStore.select(state => state.createEndUserAgreementSuccess);
+    /** Holds Store Institutions list provider company API success state as observable*/
+    public institutionsListInProgress$: Observable<any> = this.componentStore.select(state => state.institutionsListInProgress);
+    /** Instance for form group*/
+    public searchForm: FormGroup;
+    /** Hold filetered item from bank list*/
+    public filteredBanks: any[] = [];
+
+    constructor(
+        private componentStore: SettingIntegrationComponentStore,
+        public dialogRef: MatDialogRef<InstitutionsListComponent>,
+        private changeDetection: ChangeDetectorRef,
+        private generalService: GeneralService,
+        private formBuilder: FormBuilder,
+        public dialog: MatDialog,
+        @Inject(MAT_DIALOG_DATA) public inputData
+    ) {
+        this.initialForm();
+    }
+
+    /**
+     * Initializes the component
+     *
+     * @memberof InstitutionsListComponent
+     */
+    public ngOnInit(): void {
+        this.localeData = this.inputData?.localeData;
+        this.commonLocaleData = this.inputData?.commonLocaleData;
+        this.getAllInstitutionsList();
+        this.institutionsList$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.institutions = response.results;
+                this.filteredBanks = response.results;
+                this.changeDetection.detectChanges();
+            }
+        });
+
+        this.createEndUserAgreementSuccess$.pipe(takeUntil(this.destroyed$)).subscribe(response => {
+            if (response) {
+                this.dialog?.closeAll();
+                this.openWindow(response.link);
+                this.dialogRef.close(response?.reference);
```

Comment:
why dialog close all is used??

since all dialogs are closed above, are you still getting this reference number in afterClosed event?

Reply:
DONE